internal/fleet/enrollment_tokens_data_source_test.go

```diff
@@ -0,0 +1,39 @@
+package fleet_test
+
+import (
+	"testing"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/acctest"
+	"github.com/elastic/terraform-provider-elasticstack/internal/versionutils"
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+var minVersionEnrollmentTokens = version.Must(version.NewVersion("8.6.0"))
+
+func TestAccDataSourceEnrollmentTokens(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV5ProviderFactories: acctest.Providers,
+		Steps: []resource.TestStep{
+			{
+				SkipFunc: versionutils.CheckIfVersionIsUnsupported(minVersionEnrollmentTokens),
+				Config:   testAccDataSourceEnrollmentToken,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.elasticstack_fleet_enrollment_tokens.test", "policy_id", "223b1bf8-240f-463f-8466-5062670d0754"),
```

Comment:
If there going to be a resource to create enrolment tokens? If not then maybe we should check that the returned list is empty? This test just feels a bit like "yay it doesn't blow up" right now.

Reply:
> An enrollment token is automatically generated for an agent policy, so the most typical case right now is retrieving the token associated with that policy.

I see you're doing that in the `agent_policy_resource_test` file. I think we should do the same here, and assert that the data source retrieves a matching enrolment token. WDYT? Alternatively we could create a policy, and then get all the enrolment tokens and verify that at least one for the managed policy exists. 

The magic GUID feels weird here, perhaps there's some context I'm missing there and that GUID is meaningful in some way?

Reply:
> Alternatively we could create a policy, and then get all the enrolment tokens and verify that at least one for the managed policy exists.

I think this is the better option, at least we are testing something real and will work in every circumstance. It's no longer testing in isolation (we are involving agent policy now), but I'm fine with that given the circumstances. I'll go ahead and add that.

> The magic GUID feels weird here, perhaps there's some context I'm missing there and that GUID is meaningful in some way?

Nope, it was just a random UUID and has no special meaning.